pkg/cloudprovider/aws/cloudprovider.go

```diff
@@ -140,7 +139,38 @@ func (c *CloudProvider) GetInstanceTypes(ctx context.Context, provisioner *v1alp
 	if err != nil {
 		return nil, err
 	}
-	return c.instanceTypeProvider.Get(ctx, aws)
+	instanceTypes, err := c.instanceTypeProvider.Get(ctx, aws)
```

Comment:
The responsibilities of this top level interface are more or less limited to delegating our requests to other components (e.g. instanceTypeProvider). Does it make more sense to encapsulate this there alongside instance-type specific logic?

Reply:
The instance type provider maintains a cache of instance types, while this does provisioner specific filtering.  Seems best to keep the filtering here at the CP and let instance type provider be just a generic cached fetcher of instance types.